// @prisma/client/browser.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file should be your main import to use Prisma-related types and utilities in a browser. 
 * Use it to get access to models, enums, and input types.
 * 
 * This file does not contain a `PrismaClient` class, nor several other helpers that are intended as server-side only.
 * See `client.ts` for the standard, server-side entry point.
 *
 * 🟢 You can import this file directly.
 */

import * as Prisma from './internal/prismaNamespaceBrowser'
export { Prisma }
export * as $Enums from './enums'
export * from './enums';
/**
 * Model User
 * 
 */
export type User = Prisma.UserModel
/**
 * Model Workspace
 * 
 */
export type Workspace = Prisma.WorkspaceModel
/**
 * Model WorkspaceMember
 * 
 */
export type WorkspaceMember = Prisma.WorkspaceMemberModel
/**
 * Model Project
 * 
 */
export type Project = Prisma.ProjectModel
/**
 * Model ProjectAccess
 * 
 */
export type ProjectAccess = Prisma.ProjectAccessModel
/**
 * Model Task
 * 
 */
export type Task = Prisma.TaskModel
/**
 * Model Activity
 * 
 */
export type Activity = Prisma.ActivityModel
/**
 * Model Comment
 * 
 */
export type Comment = Prisma.CommentModel
/**
 * Model Documentation
 * 
 */
export type Documentation = Prisma.DocumentationModel
/**
 * Model File
 * 
 */
export type File = Prisma.FileModel
/**
 * Model Subscription
 * 
 */
export type Subscription = Prisma.SubscriptionModel
