// @prisma/client/models/Documentation.ts

/* !!! This is code generated by Prisma. Do not edit directly. !!! */
/* eslint-disable */
// biome-ignore-all lint: generated file
// @ts-nocheck 
/*
 * This file exports the `Documentation` model and its related types.
 *
 * 🟢 You can import this file directly.
 */
import type * as runtime from "@prisma/client/runtime/client"
import type * as $Enums from "../enums"
import type * as Prisma from "../internal/prismaNamespace"

/**
 * Model Documentation
 * 
 */
export type DocumentationModel = runtime.Types.Result.DefaultSelection<Prisma.$DocumentationPayload>

export type AggregateDocumentation = {
  _count: DocumentationCountAggregateOutputType | null
  _min: DocumentationMinAggregateOutputType | null
  _max: DocumentationMaxAggregateOutputType | null
}

export type DocumentationMinAggregateOutputType = {
  id: string | null
  projectId: string | null
  content: string | null
  createdAt: Date | null
  updatedAt: Date | null
  updatedBy: string | null
}

export type DocumentationMaxAggregateOutputType = {
  id: string | null
  projectId: string | null
  content: string | null
  createdAt: Date | null
  updatedAt: Date | null
  updatedBy: string | null
}

export type DocumentationCountAggregateOutputType = {
  id: number
  projectId: number
  content: number
  createdAt: number
  updatedAt: number
  updatedBy: number
  _all: number
}


export type DocumentationMinAggregateInputType = {
  id?: true
  projectId?: true
  content?: true
  createdAt?: true
  updatedAt?: true
  updatedBy?: true
}

export type DocumentationMaxAggregateInputType = {
  id?: true
  projectId?: true
  content?: true
  createdAt?: true
  updatedAt?: true
  updatedBy?: true
}

export type DocumentationCountAggregateInputType = {
  id?: true
  projectId?: true
  content?: true
  createdAt?: true
  updatedAt?: true
  updatedBy?: true
  _all?: true
}

export type DocumentationAggregateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Documentation to aggregate.
   */
  where?: Prisma.DocumentationWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Documentations to fetch.
   */
  orderBy?: Prisma.DocumentationOrderByWithRelationInput | Prisma.DocumentationOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the start position
   */
  cursor?: Prisma.DocumentationWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Documentations from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Documentations.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Count returned Documentations
  **/
  _count?: true | DocumentationCountAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the minimum value
  **/
  _min?: DocumentationMinAggregateInputType
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/aggregations Aggregation Docs}
   * 
   * Select which fields to find the maximum value
  **/
  _max?: DocumentationMaxAggregateInputType
}

export type GetDocumentationAggregateType<T extends DocumentationAggregateArgs> = {
      [P in keyof T & keyof AggregateDocumentation]: P extends '_count' | 'count'
    ? T[P] extends true
      ? number
      : Prisma.GetScalarType<T[P], AggregateDocumentation[P]>
    : Prisma.GetScalarType<T[P], AggregateDocumentation[P]>
}




export type DocumentationGroupByArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  where?: Prisma.DocumentationWhereInput
  orderBy?: Prisma.DocumentationOrderByWithAggregationInput | Prisma.DocumentationOrderByWithAggregationInput[]
  by: Prisma.DocumentationScalarFieldEnum[] | Prisma.DocumentationScalarFieldEnum
  having?: Prisma.DocumentationScalarWhereWithAggregatesInput
  take?: number
  skip?: number
  _count?: DocumentationCountAggregateInputType | true
  _min?: DocumentationMinAggregateInputType
  _max?: DocumentationMaxAggregateInputType
}

export type DocumentationGroupByOutputType = {
  id: string
  projectId: string
  content: string
  createdAt: Date
  updatedAt: Date
  updatedBy: string | null
  _count: DocumentationCountAggregateOutputType | null
  _min: DocumentationMinAggregateOutputType | null
  _max: DocumentationMaxAggregateOutputType | null
}

type GetDocumentationGroupByPayload<T extends DocumentationGroupByArgs> = Prisma.PrismaPromise<
  Array<
    Prisma.PickEnumerable<DocumentationGroupByOutputType, T['by']> &
      {
        [P in ((keyof T) & (keyof DocumentationGroupByOutputType))]: P extends '_count'
          ? T[P] extends boolean
            ? number
            : Prisma.GetScalarType<T[P], DocumentationGroupByOutputType[P]>
          : Prisma.GetScalarType<T[P], DocumentationGroupByOutputType[P]>
      }
    >
  >



export type DocumentationWhereInput = {
  AND?: Prisma.DocumentationWhereInput | Prisma.DocumentationWhereInput[]
  OR?: Prisma.DocumentationWhereInput[]
  NOT?: Prisma.DocumentationWhereInput | Prisma.DocumentationWhereInput[]
  id?: Prisma.StringFilter<"Documentation"> | string
  projectId?: Prisma.StringFilter<"Documentation"> | string
  content?: Prisma.StringFilter<"Documentation"> | string
  createdAt?: Prisma.DateTimeFilter<"Documentation"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"Documentation"> | Date | string
  updatedBy?: Prisma.StringNullableFilter<"Documentation"> | string | null
  project?: Prisma.XOR<Prisma.ProjectScalarRelationFilter, Prisma.ProjectWhereInput>
}

export type DocumentationOrderByWithRelationInput = {
  id?: Prisma.SortOrder
  projectId?: Prisma.SortOrder
  content?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  updatedBy?: Prisma.SortOrderInput | Prisma.SortOrder
  project?: Prisma.ProjectOrderByWithRelationInput
}

export type DocumentationWhereUniqueInput = Prisma.AtLeast<{
  id?: string
  projectId?: string
  AND?: Prisma.DocumentationWhereInput | Prisma.DocumentationWhereInput[]
  OR?: Prisma.DocumentationWhereInput[]
  NOT?: Prisma.DocumentationWhereInput | Prisma.DocumentationWhereInput[]
  content?: Prisma.StringFilter<"Documentation"> | string
  createdAt?: Prisma.DateTimeFilter<"Documentation"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"Documentation"> | Date | string
  updatedBy?: Prisma.StringNullableFilter<"Documentation"> | string | null
  project?: Prisma.XOR<Prisma.ProjectScalarRelationFilter, Prisma.ProjectWhereInput>
}, "id" | "projectId">

export type DocumentationOrderByWithAggregationInput = {
  id?: Prisma.SortOrder
  projectId?: Prisma.SortOrder
  content?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  updatedBy?: Prisma.SortOrderInput | Prisma.SortOrder
  _count?: Prisma.DocumentationCountOrderByAggregateInput
  _max?: Prisma.DocumentationMaxOrderByAggregateInput
  _min?: Prisma.DocumentationMinOrderByAggregateInput
}

export type DocumentationScalarWhereWithAggregatesInput = {
  AND?: Prisma.DocumentationScalarWhereWithAggregatesInput | Prisma.DocumentationScalarWhereWithAggregatesInput[]
  OR?: Prisma.DocumentationScalarWhereWithAggregatesInput[]
  NOT?: Prisma.DocumentationScalarWhereWithAggregatesInput | Prisma.DocumentationScalarWhereWithAggregatesInput[]
  id?: Prisma.StringWithAggregatesFilter<"Documentation"> | string
  projectId?: Prisma.StringWithAggregatesFilter<"Documentation"> | string
  content?: Prisma.StringWithAggregatesFilter<"Documentation"> | string
  createdAt?: Prisma.DateTimeWithAggregatesFilter<"Documentation"> | Date | string
  updatedAt?: Prisma.DateTimeWithAggregatesFilter<"Documentation"> | Date | string
  updatedBy?: Prisma.StringNullableWithAggregatesFilter<"Documentation"> | string | null
}

export type DocumentationCreateInput = {
  id?: string
  content: string
  createdAt?: Date | string
  updatedAt?: Date | string
  updatedBy?: string | null
  project: Prisma.ProjectCreateNestedOneWithoutDocumentationInput
}

export type DocumentationUncheckedCreateInput = {
  id?: string
  projectId: string
  content: string
  createdAt?: Date | string
  updatedAt?: Date | string
  updatedBy?: string | null
}

export type DocumentationUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  content?: Prisma.StringFieldUpdateOperationsInput | string
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedBy?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
  project?: Prisma.ProjectUpdateOneRequiredWithoutDocumentationNestedInput
}

export type DocumentationUncheckedUpdateInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  projectId?: Prisma.StringFieldUpdateOperationsInput | string
  content?: Prisma.StringFieldUpdateOperationsInput | string
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedBy?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type DocumentationCreateManyInput = {
  id?: string
  projectId: string
  content: string
  createdAt?: Date | string
  updatedAt?: Date | string
  updatedBy?: string | null
}

export type DocumentationUpdateManyMutationInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  content?: Prisma.StringFieldUpdateOperationsInput | string
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedBy?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type DocumentationUncheckedUpdateManyInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  projectId?: Prisma.StringFieldUpdateOperationsInput | string
  content?: Prisma.StringFieldUpdateOperationsInput | string
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedBy?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type DocumentationListRelationFilter = {
  every?: Prisma.DocumentationWhereInput
  some?: Prisma.DocumentationWhereInput
  none?: Prisma.DocumentationWhereInput
}

export type DocumentationOrderByRelationAggregateInput = {
  _count?: Prisma.SortOrder
}

export type DocumentationCountOrderByAggregateInput = {
  id?: Prisma.SortOrder
  projectId?: Prisma.SortOrder
  content?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  updatedBy?: Prisma.SortOrder
}

export type DocumentationMaxOrderByAggregateInput = {
  id?: Prisma.SortOrder
  projectId?: Prisma.SortOrder
  content?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  updatedBy?: Prisma.SortOrder
}

export type DocumentationMinOrderByAggregateInput = {
  id?: Prisma.SortOrder
  projectId?: Prisma.SortOrder
  content?: Prisma.SortOrder
  createdAt?: Prisma.SortOrder
  updatedAt?: Prisma.SortOrder
  updatedBy?: Prisma.SortOrder
}

export type DocumentationCreateNestedManyWithoutProjectInput = {
  create?: Prisma.XOR<Prisma.DocumentationCreateWithoutProjectInput, Prisma.DocumentationUncheckedCreateWithoutProjectInput> | Prisma.DocumentationCreateWithoutProjectInput[] | Prisma.DocumentationUncheckedCreateWithoutProjectInput[]
  connectOrCreate?: Prisma.DocumentationCreateOrConnectWithoutProjectInput | Prisma.DocumentationCreateOrConnectWithoutProjectInput[]
  createMany?: Prisma.DocumentationCreateManyProjectInputEnvelope
  connect?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
}

export type DocumentationUncheckedCreateNestedManyWithoutProjectInput = {
  create?: Prisma.XOR<Prisma.DocumentationCreateWithoutProjectInput, Prisma.DocumentationUncheckedCreateWithoutProjectInput> | Prisma.DocumentationCreateWithoutProjectInput[] | Prisma.DocumentationUncheckedCreateWithoutProjectInput[]
  connectOrCreate?: Prisma.DocumentationCreateOrConnectWithoutProjectInput | Prisma.DocumentationCreateOrConnectWithoutProjectInput[]
  createMany?: Prisma.DocumentationCreateManyProjectInputEnvelope
  connect?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
}

export type DocumentationUpdateManyWithoutProjectNestedInput = {
  create?: Prisma.XOR<Prisma.DocumentationCreateWithoutProjectInput, Prisma.DocumentationUncheckedCreateWithoutProjectInput> | Prisma.DocumentationCreateWithoutProjectInput[] | Prisma.DocumentationUncheckedCreateWithoutProjectInput[]
  connectOrCreate?: Prisma.DocumentationCreateOrConnectWithoutProjectInput | Prisma.DocumentationCreateOrConnectWithoutProjectInput[]
  upsert?: Prisma.DocumentationUpsertWithWhereUniqueWithoutProjectInput | Prisma.DocumentationUpsertWithWhereUniqueWithoutProjectInput[]
  createMany?: Prisma.DocumentationCreateManyProjectInputEnvelope
  set?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  disconnect?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  delete?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  connect?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  update?: Prisma.DocumentationUpdateWithWhereUniqueWithoutProjectInput | Prisma.DocumentationUpdateWithWhereUniqueWithoutProjectInput[]
  updateMany?: Prisma.DocumentationUpdateManyWithWhereWithoutProjectInput | Prisma.DocumentationUpdateManyWithWhereWithoutProjectInput[]
  deleteMany?: Prisma.DocumentationScalarWhereInput | Prisma.DocumentationScalarWhereInput[]
}

export type DocumentationUncheckedUpdateManyWithoutProjectNestedInput = {
  create?: Prisma.XOR<Prisma.DocumentationCreateWithoutProjectInput, Prisma.DocumentationUncheckedCreateWithoutProjectInput> | Prisma.DocumentationCreateWithoutProjectInput[] | Prisma.DocumentationUncheckedCreateWithoutProjectInput[]
  connectOrCreate?: Prisma.DocumentationCreateOrConnectWithoutProjectInput | Prisma.DocumentationCreateOrConnectWithoutProjectInput[]
  upsert?: Prisma.DocumentationUpsertWithWhereUniqueWithoutProjectInput | Prisma.DocumentationUpsertWithWhereUniqueWithoutProjectInput[]
  createMany?: Prisma.DocumentationCreateManyProjectInputEnvelope
  set?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  disconnect?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  delete?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  connect?: Prisma.DocumentationWhereUniqueInput | Prisma.DocumentationWhereUniqueInput[]
  update?: Prisma.DocumentationUpdateWithWhereUniqueWithoutProjectInput | Prisma.DocumentationUpdateWithWhereUniqueWithoutProjectInput[]
  updateMany?: Prisma.DocumentationUpdateManyWithWhereWithoutProjectInput | Prisma.DocumentationUpdateManyWithWhereWithoutProjectInput[]
  deleteMany?: Prisma.DocumentationScalarWhereInput | Prisma.DocumentationScalarWhereInput[]
}

export type DocumentationCreateWithoutProjectInput = {
  id?: string
  content: string
  createdAt?: Date | string
  updatedAt?: Date | string
  updatedBy?: string | null
}

export type DocumentationUncheckedCreateWithoutProjectInput = {
  id?: string
  content: string
  createdAt?: Date | string
  updatedAt?: Date | string
  updatedBy?: string | null
}

export type DocumentationCreateOrConnectWithoutProjectInput = {
  where: Prisma.DocumentationWhereUniqueInput
  create: Prisma.XOR<Prisma.DocumentationCreateWithoutProjectInput, Prisma.DocumentationUncheckedCreateWithoutProjectInput>
}

export type DocumentationCreateManyProjectInputEnvelope = {
  data: Prisma.DocumentationCreateManyProjectInput | Prisma.DocumentationCreateManyProjectInput[]
  skipDuplicates?: boolean
}

export type DocumentationUpsertWithWhereUniqueWithoutProjectInput = {
  where: Prisma.DocumentationWhereUniqueInput
  update: Prisma.XOR<Prisma.DocumentationUpdateWithoutProjectInput, Prisma.DocumentationUncheckedUpdateWithoutProjectInput>
  create: Prisma.XOR<Prisma.DocumentationCreateWithoutProjectInput, Prisma.DocumentationUncheckedCreateWithoutProjectInput>
}

export type DocumentationUpdateWithWhereUniqueWithoutProjectInput = {
  where: Prisma.DocumentationWhereUniqueInput
  data: Prisma.XOR<Prisma.DocumentationUpdateWithoutProjectInput, Prisma.DocumentationUncheckedUpdateWithoutProjectInput>
}

export type DocumentationUpdateManyWithWhereWithoutProjectInput = {
  where: Prisma.DocumentationScalarWhereInput
  data: Prisma.XOR<Prisma.DocumentationUpdateManyMutationInput, Prisma.DocumentationUncheckedUpdateManyWithoutProjectInput>
}

export type DocumentationScalarWhereInput = {
  AND?: Prisma.DocumentationScalarWhereInput | Prisma.DocumentationScalarWhereInput[]
  OR?: Prisma.DocumentationScalarWhereInput[]
  NOT?: Prisma.DocumentationScalarWhereInput | Prisma.DocumentationScalarWhereInput[]
  id?: Prisma.StringFilter<"Documentation"> | string
  projectId?: Prisma.StringFilter<"Documentation"> | string
  content?: Prisma.StringFilter<"Documentation"> | string
  createdAt?: Prisma.DateTimeFilter<"Documentation"> | Date | string
  updatedAt?: Prisma.DateTimeFilter<"Documentation"> | Date | string
  updatedBy?: Prisma.StringNullableFilter<"Documentation"> | string | null
}

export type DocumentationCreateManyProjectInput = {
  id?: string
  content: string
  createdAt?: Date | string
  updatedAt?: Date | string
  updatedBy?: string | null
}

export type DocumentationUpdateWithoutProjectInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  content?: Prisma.StringFieldUpdateOperationsInput | string
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedBy?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type DocumentationUncheckedUpdateWithoutProjectInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  content?: Prisma.StringFieldUpdateOperationsInput | string
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedBy?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}

export type DocumentationUncheckedUpdateManyWithoutProjectInput = {
  id?: Prisma.StringFieldUpdateOperationsInput | string
  content?: Prisma.StringFieldUpdateOperationsInput | string
  createdAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedAt?: Prisma.DateTimeFieldUpdateOperationsInput | Date | string
  updatedBy?: Prisma.NullableStringFieldUpdateOperationsInput | string | null
}



export type DocumentationSelect<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  projectId?: boolean
  content?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  updatedBy?: boolean
  project?: boolean | Prisma.ProjectDefaultArgs<ExtArgs>
}, ExtArgs["result"]["documentation"]>

export type DocumentationSelectCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  projectId?: boolean
  content?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  updatedBy?: boolean
  project?: boolean | Prisma.ProjectDefaultArgs<ExtArgs>
}, ExtArgs["result"]["documentation"]>

export type DocumentationSelectUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetSelect<{
  id?: boolean
  projectId?: boolean
  content?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  updatedBy?: boolean
  project?: boolean | Prisma.ProjectDefaultArgs<ExtArgs>
}, ExtArgs["result"]["documentation"]>

export type DocumentationSelectScalar = {
  id?: boolean
  projectId?: boolean
  content?: boolean
  createdAt?: boolean
  updatedAt?: boolean
  updatedBy?: boolean
}

export type DocumentationOmit<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = runtime.Types.Extensions.GetOmit<"id" | "projectId" | "content" | "createdAt" | "updatedAt" | "updatedBy", ExtArgs["result"]["documentation"]>
export type DocumentationInclude<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  project?: boolean | Prisma.ProjectDefaultArgs<ExtArgs>
}
export type DocumentationIncludeCreateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  project?: boolean | Prisma.ProjectDefaultArgs<ExtArgs>
}
export type DocumentationIncludeUpdateManyAndReturn<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  project?: boolean | Prisma.ProjectDefaultArgs<ExtArgs>
}

export type $DocumentationPayload<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  name: "Documentation"
  objects: {
    project: Prisma.$ProjectPayload<ExtArgs>
  }
  scalars: runtime.Types.Extensions.GetPayloadResult<{
    id: string
    projectId: string
    content: string
    createdAt: Date
    updatedAt: Date
    updatedBy: string | null
  }, ExtArgs["result"]["documentation"]>
  composites: {}
}

export type DocumentationGetPayload<S extends boolean | null | undefined | DocumentationDefaultArgs> = runtime.Types.Result.GetResult<Prisma.$DocumentationPayload, S>

export type DocumentationCountArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> =
  Omit<DocumentationFindManyArgs, 'select' | 'include' | 'distinct' | 'omit'> & {
    select?: DocumentationCountAggregateInputType | true
  }

export interface DocumentationDelegate<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> {
  [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['Documentation'], meta: { name: 'Documentation' } }
  /**
   * Find zero or one Documentation that matches the filter.
   * @param {DocumentationFindUniqueArgs} args - Arguments to find a Documentation
   * @example
   * // Get one Documentation
   * const documentation = await prisma.documentation.findUnique({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUnique<T extends DocumentationFindUniqueArgs>(args: Prisma.SelectSubset<T, DocumentationFindUniqueArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "findUnique", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find one Documentation that matches the filter or throw an error with `error.code='P2025'`
   * if no matches were found.
   * @param {DocumentationFindUniqueOrThrowArgs} args - Arguments to find a Documentation
   * @example
   * // Get one Documentation
   * const documentation = await prisma.documentation.findUniqueOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findUniqueOrThrow<T extends DocumentationFindUniqueOrThrowArgs>(args: Prisma.SelectSubset<T, DocumentationFindUniqueOrThrowArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Documentation that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {DocumentationFindFirstArgs} args - Arguments to find a Documentation
   * @example
   * // Get one Documentation
   * const documentation = await prisma.documentation.findFirst({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirst<T extends DocumentationFindFirstArgs>(args?: Prisma.SelectSubset<T, DocumentationFindFirstArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "findFirst", GlobalOmitOptions> | null, null, ExtArgs, GlobalOmitOptions>

  /**
   * Find the first Documentation that matches the filter or
   * throw `PrismaKnownClientError` with `P2025` code if no matches were found.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {DocumentationFindFirstOrThrowArgs} args - Arguments to find a Documentation
   * @example
   * // Get one Documentation
   * const documentation = await prisma.documentation.findFirstOrThrow({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   */
  findFirstOrThrow<T extends DocumentationFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<T, DocumentationFindFirstOrThrowArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "findFirstOrThrow", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Find zero or more Documentations that matches the filter.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {DocumentationFindManyArgs} args - Arguments to filter and select certain fields only.
   * @example
   * // Get all Documentations
   * const documentations = await prisma.documentation.findMany()
   * 
   * // Get first 10 Documentations
   * const documentations = await prisma.documentation.findMany({ take: 10 })
   * 
   * // Only select the `id`
   * const documentationWithIdOnly = await prisma.documentation.findMany({ select: { id: true } })
   * 
   */
  findMany<T extends DocumentationFindManyArgs>(args?: Prisma.SelectSubset<T, DocumentationFindManyArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "findMany", GlobalOmitOptions>>

  /**
   * Create a Documentation.
   * @param {DocumentationCreateArgs} args - Arguments to create a Documentation.
   * @example
   * // Create one Documentation
   * const Documentation = await prisma.documentation.create({
   *   data: {
   *     // ... data to create a Documentation
   *   }
   * })
   * 
   */
  create<T extends DocumentationCreateArgs>(args: Prisma.SelectSubset<T, DocumentationCreateArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "create", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Create many Documentations.
   * @param {DocumentationCreateManyArgs} args - Arguments to create many Documentations.
   * @example
   * // Create many Documentations
   * const documentation = await prisma.documentation.createMany({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   *     
   */
  createMany<T extends DocumentationCreateManyArgs>(args?: Prisma.SelectSubset<T, DocumentationCreateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Create many Documentations and returns the data saved in the database.
   * @param {DocumentationCreateManyAndReturnArgs} args - Arguments to create many Documentations.
   * @example
   * // Create many Documentations
   * const documentation = await prisma.documentation.createManyAndReturn({
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Create many Documentations and only return the `id`
   * const documentationWithIdOnly = await prisma.documentation.createManyAndReturn({
   *   select: { id: true },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  createManyAndReturn<T extends DocumentationCreateManyAndReturnArgs>(args?: Prisma.SelectSubset<T, DocumentationCreateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "createManyAndReturn", GlobalOmitOptions>>

  /**
   * Delete a Documentation.
   * @param {DocumentationDeleteArgs} args - Arguments to delete one Documentation.
   * @example
   * // Delete one Documentation
   * const Documentation = await prisma.documentation.delete({
   *   where: {
   *     // ... filter to delete one Documentation
   *   }
   * })
   * 
   */
  delete<T extends DocumentationDeleteArgs>(args: Prisma.SelectSubset<T, DocumentationDeleteArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "delete", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Update one Documentation.
   * @param {DocumentationUpdateArgs} args - Arguments to update one Documentation.
   * @example
   * // Update one Documentation
   * const documentation = await prisma.documentation.update({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  update<T extends DocumentationUpdateArgs>(args: Prisma.SelectSubset<T, DocumentationUpdateArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "update", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>

  /**
   * Delete zero or more Documentations.
   * @param {DocumentationDeleteManyArgs} args - Arguments to filter Documentations to delete.
   * @example
   * // Delete a few Documentations
   * const { count } = await prisma.documentation.deleteMany({
   *   where: {
   *     // ... provide filter here
   *   }
   * })
   * 
   */
  deleteMany<T extends DocumentationDeleteManyArgs>(args?: Prisma.SelectSubset<T, DocumentationDeleteManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Documentations.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {DocumentationUpdateManyArgs} args - Arguments to update one or more rows.
   * @example
   * // Update many Documentations
   * const documentation = await prisma.documentation.updateMany({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: {
   *     // ... provide data here
   *   }
   * })
   * 
   */
  updateMany<T extends DocumentationUpdateManyArgs>(args: Prisma.SelectSubset<T, DocumentationUpdateManyArgs<ExtArgs>>): Prisma.PrismaPromise<Prisma.BatchPayload>

  /**
   * Update zero or more Documentations and returns the data updated in the database.
   * @param {DocumentationUpdateManyAndReturnArgs} args - Arguments to update many Documentations.
   * @example
   * // Update many Documentations
   * const documentation = await prisma.documentation.updateManyAndReturn({
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * 
   * // Update zero or more Documentations and only return the `id`
   * const documentationWithIdOnly = await prisma.documentation.updateManyAndReturn({
   *   select: { id: true },
   *   where: {
   *     // ... provide filter here
   *   },
   *   data: [
   *     // ... provide data here
   *   ]
   * })
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * 
   */
  updateManyAndReturn<T extends DocumentationUpdateManyAndReturnArgs>(args: Prisma.SelectSubset<T, DocumentationUpdateManyAndReturnArgs<ExtArgs>>): Prisma.PrismaPromise<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "updateManyAndReturn", GlobalOmitOptions>>

  /**
   * Create or update one Documentation.
   * @param {DocumentationUpsertArgs} args - Arguments to update or create a Documentation.
   * @example
   * // Update or create a Documentation
   * const documentation = await prisma.documentation.upsert({
   *   create: {
   *     // ... data to create a Documentation
   *   },
   *   update: {
   *     // ... in case it already exists, update
   *   },
   *   where: {
   *     // ... the filter for the Documentation we want to update
   *   }
   * })
   */
  upsert<T extends DocumentationUpsertArgs>(args: Prisma.SelectSubset<T, DocumentationUpsertArgs<ExtArgs>>): Prisma.Prisma__DocumentationClient<runtime.Types.Result.GetResult<Prisma.$DocumentationPayload<ExtArgs>, T, "upsert", GlobalOmitOptions>, never, ExtArgs, GlobalOmitOptions>


  /**
   * Count the number of Documentations.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {DocumentationCountArgs} args - Arguments to filter Documentations to count.
   * @example
   * // Count the number of Documentations
   * const count = await prisma.documentation.count({
   *   where: {
   *     // ... the filter for the Documentations we want to count
   *   }
   * })
  **/
  count<T extends DocumentationCountArgs>(
    args?: Prisma.Subset<T, DocumentationCountArgs>,
  ): Prisma.PrismaPromise<
    T extends runtime.Types.Utils.Record<'select', any>
      ? T['select'] extends true
        ? number
        : Prisma.GetScalarType<T['select'], DocumentationCountAggregateOutputType>
      : number
  >

  /**
   * Allows you to perform aggregations operations on a Documentation.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {DocumentationAggregateArgs} args - Select which aggregations you would like to apply and on what fields.
   * @example
   * // Ordered by age ascending
   * // Where email contains prisma.io
   * // Limited to the 10 users
   * const aggregations = await prisma.user.aggregate({
   *   _avg: {
   *     age: true,
   *   },
   *   where: {
   *     email: {
   *       contains: "prisma.io",
   *     },
   *   },
   *   orderBy: {
   *     age: "asc",
   *   },
   *   take: 10,
   * })
  **/
  aggregate<T extends DocumentationAggregateArgs>(args: Prisma.Subset<T, DocumentationAggregateArgs>): Prisma.PrismaPromise<GetDocumentationAggregateType<T>>

  /**
   * Group by Documentation.
   * Note, that providing `undefined` is treated as the value not being there.
   * Read more here: https://pris.ly/d/null-undefined
   * @param {DocumentationGroupByArgs} args - Group by arguments.
   * @example
   * // Group by city, order by createdAt, get count
   * const result = await prisma.user.groupBy({
   *   by: ['city', 'createdAt'],
   *   orderBy: {
   *     createdAt: true
   *   },
   *   _count: {
   *     _all: true
   *   },
   * })
   * 
  **/
  groupBy<
    T extends DocumentationGroupByArgs,
    HasSelectOrTake extends Prisma.Or<
      Prisma.Extends<'skip', Prisma.Keys<T>>,
      Prisma.Extends<'take', Prisma.Keys<T>>
    >,
    OrderByArg extends Prisma.True extends HasSelectOrTake
      ? { orderBy: DocumentationGroupByArgs['orderBy'] }
      : { orderBy?: DocumentationGroupByArgs['orderBy'] },
    OrderFields extends Prisma.ExcludeUnderscoreKeys<Prisma.Keys<Prisma.MaybeTupleToUnion<T['orderBy']>>>,
    ByFields extends Prisma.MaybeTupleToUnion<T['by']>,
    ByValid extends Prisma.Has<ByFields, OrderFields>,
    HavingFields extends Prisma.GetHavingFields<T['having']>,
    HavingValid extends Prisma.Has<ByFields, HavingFields>,
    ByEmpty extends T['by'] extends never[] ? Prisma.True : Prisma.False,
    InputErrors extends ByEmpty extends Prisma.True
    ? `Error: "by" must not be empty.`
    : HavingValid extends Prisma.False
    ? {
        [P in HavingFields]: P extends ByFields
          ? never
          : P extends string
          ? `Error: Field "${P}" used in "having" needs to be provided in "by".`
          : [
              Error,
              'Field ',
              P,
              ` in "having" needs to be provided in "by"`,
            ]
      }[HavingFields]
    : 'take' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "take", you also need to provide "orderBy"'
    : 'skip' extends Prisma.Keys<T>
    ? 'orderBy' extends Prisma.Keys<T>
      ? ByValid extends Prisma.True
        ? {}
        : {
            [P in OrderFields]: P extends ByFields
              ? never
              : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
          }[OrderFields]
      : 'Error: If you provide "skip", you also need to provide "orderBy"'
    : ByValid extends Prisma.True
    ? {}
    : {
        [P in OrderFields]: P extends ByFields
          ? never
          : `Error: Field "${P}" in "orderBy" needs to be provided in "by"`
      }[OrderFields]
  >(args: Prisma.SubsetIntersection<T, DocumentationGroupByArgs, OrderByArg> & InputErrors): {} extends InputErrors ? GetDocumentationGroupByPayload<T> : Prisma.PrismaPromise<InputErrors>
/**
 * Fields of the Documentation model
 */
readonly fields: DocumentationFieldRefs;
}

/**
 * The delegate class that acts as a "Promise-like" for Documentation.
 * Why is this prefixed with `Prisma__`?
 * Because we want to prevent naming conflicts as mentioned in
 * https://github.com/prisma/prisma-client-js/issues/707
 */
export interface Prisma__DocumentationClient<T, Null = never, ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs, GlobalOmitOptions = {}> extends Prisma.PrismaPromise<T> {
  readonly [Symbol.toStringTag]: "PrismaPromise"
  project<T extends Prisma.ProjectDefaultArgs<ExtArgs> = {}>(args?: Prisma.Subset<T, Prisma.ProjectDefaultArgs<ExtArgs>>): Prisma.Prisma__ProjectClient<runtime.Types.Result.GetResult<Prisma.$ProjectPayload<ExtArgs>, T, "findUniqueOrThrow", GlobalOmitOptions> | Null, Null, ExtArgs, GlobalOmitOptions>
  /**
   * Attaches callbacks for the resolution and/or rejection of the Promise.
   * @param onfulfilled The callback to execute when the Promise is resolved.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of which ever callback is executed.
   */
  then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): runtime.Types.Utils.JsPromise<TResult1 | TResult2>
  /**
   * Attaches a callback for only the rejection of the Promise.
   * @param onrejected The callback to execute when the Promise is rejected.
   * @returns A Promise for the completion of the callback.
   */
  catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): runtime.Types.Utils.JsPromise<T | TResult>
  /**
   * Attaches a callback that is invoked when the Promise is settled (fulfilled or rejected). The
   * resolved value cannot be modified from the callback.
   * @param onfinally The callback to execute when the Promise is settled (fulfilled or rejected).
   * @returns A Promise for the completion of the callback.
   */
  finally(onfinally?: (() => void) | undefined | null): runtime.Types.Utils.JsPromise<T>
}




/**
 * Fields of the Documentation model
 */
export interface DocumentationFieldRefs {
  readonly id: Prisma.FieldRef<"Documentation", 'String'>
  readonly projectId: Prisma.FieldRef<"Documentation", 'String'>
  readonly content: Prisma.FieldRef<"Documentation", 'String'>
  readonly createdAt: Prisma.FieldRef<"Documentation", 'DateTime'>
  readonly updatedAt: Prisma.FieldRef<"Documentation", 'DateTime'>
  readonly updatedBy: Prisma.FieldRef<"Documentation", 'String'>
}
    

// Custom InputTypes
/**
 * Documentation findUnique
 */
export type DocumentationFindUniqueArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * Filter, which Documentation to fetch.
   */
  where: Prisma.DocumentationWhereUniqueInput
}

/**
 * Documentation findUniqueOrThrow
 */
export type DocumentationFindUniqueOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * Filter, which Documentation to fetch.
   */
  where: Prisma.DocumentationWhereUniqueInput
}

/**
 * Documentation findFirst
 */
export type DocumentationFindFirstArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * Filter, which Documentation to fetch.
   */
  where?: Prisma.DocumentationWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Documentations to fetch.
   */
  orderBy?: Prisma.DocumentationOrderByWithRelationInput | Prisma.DocumentationOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Documentations.
   */
  cursor?: Prisma.DocumentationWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Documentations from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Documentations.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Documentations.
   */
  distinct?: Prisma.DocumentationScalarFieldEnum | Prisma.DocumentationScalarFieldEnum[]
}

/**
 * Documentation findFirstOrThrow
 */
export type DocumentationFindFirstOrThrowArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * Filter, which Documentation to fetch.
   */
  where?: Prisma.DocumentationWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Documentations to fetch.
   */
  orderBy?: Prisma.DocumentationOrderByWithRelationInput | Prisma.DocumentationOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for searching for Documentations.
   */
  cursor?: Prisma.DocumentationWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Documentations from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Documentations.
   */
  skip?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/distinct Distinct Docs}
   * 
   * Filter by unique combinations of Documentations.
   */
  distinct?: Prisma.DocumentationScalarFieldEnum | Prisma.DocumentationScalarFieldEnum[]
}

/**
 * Documentation findMany
 */
export type DocumentationFindManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * Filter, which Documentations to fetch.
   */
  where?: Prisma.DocumentationWhereInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/sorting Sorting Docs}
   * 
   * Determine the order of Documentations to fetch.
   */
  orderBy?: Prisma.DocumentationOrderByWithRelationInput | Prisma.DocumentationOrderByWithRelationInput[]
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination Cursor Docs}
   * 
   * Sets the position for listing Documentations.
   */
  cursor?: Prisma.DocumentationWhereUniqueInput
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Take `±n` Documentations from the position of the cursor.
   */
  take?: number
  /**
   * {@link https://www.prisma.io/docs/concepts/components/prisma-client/pagination Pagination Docs}
   * 
   * Skip the first `n` Documentations.
   */
  skip?: number
  distinct?: Prisma.DocumentationScalarFieldEnum | Prisma.DocumentationScalarFieldEnum[]
}

/**
 * Documentation create
 */
export type DocumentationCreateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * The data needed to create a Documentation.
   */
  data: Prisma.XOR<Prisma.DocumentationCreateInput, Prisma.DocumentationUncheckedCreateInput>
}

/**
 * Documentation createMany
 */
export type DocumentationCreateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to create many Documentations.
   */
  data: Prisma.DocumentationCreateManyInput | Prisma.DocumentationCreateManyInput[]
  skipDuplicates?: boolean
}

/**
 * Documentation createManyAndReturn
 */
export type DocumentationCreateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelectCreateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * The data used to create many Documentations.
   */
  data: Prisma.DocumentationCreateManyInput | Prisma.DocumentationCreateManyInput[]
  skipDuplicates?: boolean
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationIncludeCreateManyAndReturn<ExtArgs> | null
}

/**
 * Documentation update
 */
export type DocumentationUpdateArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * The data needed to update a Documentation.
   */
  data: Prisma.XOR<Prisma.DocumentationUpdateInput, Prisma.DocumentationUncheckedUpdateInput>
  /**
   * Choose, which Documentation to update.
   */
  where: Prisma.DocumentationWhereUniqueInput
}

/**
 * Documentation updateMany
 */
export type DocumentationUpdateManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * The data used to update Documentations.
   */
  data: Prisma.XOR<Prisma.DocumentationUpdateManyMutationInput, Prisma.DocumentationUncheckedUpdateManyInput>
  /**
   * Filter which Documentations to update
   */
  where?: Prisma.DocumentationWhereInput
  /**
   * Limit how many Documentations to update.
   */
  limit?: number
}

/**
 * Documentation updateManyAndReturn
 */
export type DocumentationUpdateManyAndReturnArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelectUpdateManyAndReturn<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * The data used to update Documentations.
   */
  data: Prisma.XOR<Prisma.DocumentationUpdateManyMutationInput, Prisma.DocumentationUncheckedUpdateManyInput>
  /**
   * Filter which Documentations to update
   */
  where?: Prisma.DocumentationWhereInput
  /**
   * Limit how many Documentations to update.
   */
  limit?: number
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationIncludeUpdateManyAndReturn<ExtArgs> | null
}

/**
 * Documentation upsert
 */
export type DocumentationUpsertArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * The filter to search for the Documentation to update in case it exists.
   */
  where: Prisma.DocumentationWhereUniqueInput
  /**
   * In case the Documentation found by the `where` argument doesn't exist, create a new Documentation with this data.
   */
  create: Prisma.XOR<Prisma.DocumentationCreateInput, Prisma.DocumentationUncheckedCreateInput>
  /**
   * In case the Documentation was found with the provided `where` argument, update it with this data.
   */
  update: Prisma.XOR<Prisma.DocumentationUpdateInput, Prisma.DocumentationUncheckedUpdateInput>
}

/**
 * Documentation delete
 */
export type DocumentationDeleteArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
  /**
   * Filter which Documentation to delete.
   */
  where: Prisma.DocumentationWhereUniqueInput
}

/**
 * Documentation deleteMany
 */
export type DocumentationDeleteManyArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Filter which Documentations to delete
   */
  where?: Prisma.DocumentationWhereInput
  /**
   * Limit how many Documentations to delete.
   */
  limit?: number
}

/**
 * Documentation without action
 */
export type DocumentationDefaultArgs<ExtArgs extends runtime.Types.Extensions.InternalArgs = runtime.Types.Extensions.DefaultArgs> = {
  /**
   * Select specific fields to fetch from the Documentation
   */
  select?: Prisma.DocumentationSelect<ExtArgs> | null
  /**
   * Omit specific fields from the Documentation
   */
  omit?: Prisma.DocumentationOmit<ExtArgs> | null
  /**
   * Choose, which related nodes to fetch as well
   */
  include?: Prisma.DocumentationInclude<ExtArgs> | null
}
